internal/controller/oauth/oidc_provider.go

```diff
@@ -0,0 +1,103 @@
+package oauth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
+	"golang.org/x/oauth2"
+)
+
+// OIDCProvider extends ClientCredentialsTokenProvider with OIDC support
+type OIDCProvider struct {
+	tokenProvider  *ClientCredentialsTokenProvider
+	httpClient     *http.Client
+	oidcCredential *egv1a1.OIDC
+}
+
+// NewOIDCProvider creates a new OIDC-aware provider
+func NewOIDCProvider(tokenProvider *ClientCredentialsTokenProvider, oidcCredentials *egv1a1.OIDC) *OIDCProvider {
+	return &OIDCProvider{
+		tokenProvider:  tokenProvider,
+		httpClient:     &http.Client{Timeout: 30 * time.Second},
+		oidcCredential: oidcCredentials,
+	}
+}
+
+// getOIDCProviderConfig retrieves or creates OIDC config for the given issuer URL
+func (p *OIDCProvider) getOIDCProviderConfig(ctx context.Context, issuerURL string) (*oidc.ProviderConfig, *[]string, error) {
```

Comment:
i see there's no need to return a pointer to slice here
```suggestion
func (p *OIDCProvider) getOIDCProviderConfig(ctx context.Context, issuerURL string) (*oidc.ProviderConfig, []string, error) {
```